apps/io-func-wallet-solution-backend/src/infra/http/handlers/create-wallet-instance.ts

```diff
@@ -51,6 +54,7 @@ export const CreateWalletInstanceHandler = H.of((req: H.HttpRequest) =>
       pipe(
         consumeNonce(walletInstanceRequest.challenge),
         RTE.chainW(() => validateAttestation(walletInstanceRequest)),
+        RTE.chainFirstW(() => revokeAllWalletInstancesByUserId(user.id)),
         RTE.chainW(({ hardwareKey }) =>
           insertWalletInstance({
             id: walletInstanceRequest.hardwareKeyTag,
```

Comment:
Here it could happen that the deletion is successful but not the insertion, causing problems for the user who sees an unsuccessful request without the possibility of using the Wallet Instance. Can we check to see if the insertion was successful and then confirm the cancellation?

Reply:
so are you suggesting to place the revoke after the insert?

Reply:
Yes, still find some mechanism that guarantees integrity. However, in principle yes, I would do a delete after the insert.